petastorm/spark/spark_dataset_converter.py

```diff
@@ -23,72 +26,93 @@
 }
 
 
-class tf_dataset_context_manager:
-
-    def __init__(self, converter):
-        self.converter = converter
-
-    def __enter__(self) -> tf.data.Dataset:
-        return self.converter.dataset
-
-    def __exit__(self, exc_type, exc_value, exc_traceback):
-        self.converter.close()
-
-
 class SparkDatasetConverter:
     """
     The SparkDatasetConverter class manages the intermediate files when converting a SparkDataFrame to a
     Tensorflow Dataset or PyTorch DataLoader.
     """
-    def __init__(self, cache_file_path):
+    def __init__(self, cache_file_path: str, unischema: Unischema = None):
         """
         :param cache_file_path: The path to store the cache files.
+        :param unischema: Unischema that contains metadata of the dataset. TODO use this param.
         """
         self.cache_file_path = cache_file_path
+        self.unischema = unischema
         self.dataset = None
 
-    def make_tf_dataset(self) -> tf_dataset_context_manager:
+    def make_tf_dataset(self):
         reader = make_batch_reader("file://" + self.cache_file_path)
         self.dataset = make_petastorm_dataset(reader)
```

Comment:
The converter don't hold tf.data.Dataset, use tf_dataset_context_manager to hold it.